numpy/core/tests/test_shape_base.py

```diff
@@ -647,3 +647,55 @@ def test_different_ndims_depths(self, block):
                               [3., 3., 3.]]])
 
         assert_equal(result, expected)
+
+    def test_block_memory_order(self, block):
+        arr_c = np.zeros((3,)*3, order='C')
+        arr_f = np.zeros((3,)*3, order='F')
+
+        b_c = [[[arr_c, arr_c],
+                [arr_c, arr_c]],
+               [[arr_c, arr_c],
+                [arr_c, arr_c]]]
+
+        b_f = [[[arr_f, arr_f],
+                [arr_f, arr_f]],
+               [[arr_f, arr_f],
+                [arr_f, arr_f]]]
+
+        b_mixed = [[[arr_f, arr_f],
+                    [arr_c, arr_f]],
+                   [[arr_f, arr_f],
+                    [arr_f, arr_f]]]
+
+        assert block(b_c).flags['C_CONTIGUOUS']
+        assert block(b_mixed).flags['C_CONTIGUOUS']
+        assert block(b_f).flags['F_CONTIGUOUS']
+
+        # Remove the leading dimension in the 3D block
+        # this causes the shape of the input arrays to be different
+        # than the shape of the output array.
+        # Because of the new len 1 dimensions, it is difficult to
+        # infer the memory order of the array.
+        arr_c = np.zeros((3, 3), order='C')
+        arr_f = np.zeros((3, 3), order='F')
+
+        b_c = [[[arr_c, arr_c],
+                [arr_c, arr_c]],
+               [[arr_c, arr_c],
+                [arr_c, arr_c]]]
+
+        b_f = [[[arr_f, arr_f],
+                [arr_f, arr_f]],
+               [[arr_f, arr_f],
+                [arr_f, arr_f]]]
+
+        b_mixed = [[[arr_f, arr_f],
+                    [arr_c, arr_f]],
+                   [[arr_f, arr_f],
+                    [arr_f, arr_f]]]
+
+        assert block(b_c).flags['C_CONTIGUOUS']
+        assert block(b_mixed).flags['C_CONTIGUOUS']
```

Comment:
I don't think we should have a test for the mixed case, because we don't want to guarantee anything about this case either way. Future updates should feel free to change the order.

Reply:
ok.